### ISSUE TITLE
IHRA Professional Drag Racing 2005 - mVU Console Spam, Slow Performance 

### ISSUE THREAD

**Issue by MrCK1** [PCSX2/pcsx2#2388]
***PCSX2 version:***
v1.5.0-dev-2345-g230a497a1

***PCSX2 options:***
Defaults, Speedhacks Disabled

***Plugins used:***
GSdx OpenGL HW

***Description of the issue:***
This game has issues with MicroVU and runs rather poorly. The console will spam the messages shown below, and doesn't go over ~20FPS in-game when 3D geometry is displayed. SuperVU will work at full speed, but the sky is missing and polygons are misplaced - with similar console messages displayed.

***How to reproduce the issue:***
Start the game, go into a race.

***Last known version to work:***
N/A

***PC specifications:***
i5-6600K, GTX 1070, Windows 10 64-Bit

Console Sample:

```
microVU1: Cached Prog = [30653] [PC=39b8] [List=17523] (Cache=80.809%) [49.3mb]
microVU1: Prog Range Wrap [3ff0] [-1]
microVU1: Cached Prog = [30654] [PC=39b8] [List=17524] (Cache=80.811%) [49.3mb]
microVU1: Prog Range Wrap [3ff0] [-1]
microVU1: Cached Prog = [30655] [PC=39b8] [List=17525] (Cache=80.814%) [49.3mb]
microVU1: Prog Range Wrap [3ff0] [-1]
microVU1: Cached Prog = [30656] [PC=39b8] [List=17526] (Cache=80.817%) [49.3mb]
microVU1: Prog Range Wrap [3ff0] [-1]
microVU1: Cached Prog = [30657] [PC=39b8] [List=17527] (Cache=80.819%) [49.3mb]
microVU1: Prog Range Wrap [3ff0] [-1]
microVU1: Cached Prog = [30658] [PC=39b8] [List=17528] (Cache=80.822%) [49.3mb]
microVU1: Prog Range Wrap [3ff0] [-1]
microVU1: Cached Prog = [30659] [PC=39d8] [List=4391] (Cache=80.824%) [49.3mb]
microVU1: Prog Range Wrap [3ff0] [-1]
microVU1: Cached Prog = [30660] [PC=39d8] [List=4392] (Cache=80.827%) [49.3mb]
microVU1: Prog Range Wrap [3ff0] [-1]
microVU1: Cached Prog = [30661] [PC=39c8] [List=8709] (Cache=80.829%) [49.3mb]
microVU1: Prog Range Wrap [3ff0] [-1]
microVU1: Cached Prog = [30662] [PC=39c8] [List=8710] (Cache=80.832%) [49.3mb]
microVU1: Prog Range Wrap [3ff0] [-1]
microVU1: Cached Prog = [30663] [PC=39c8] [List=8711] (Cache=80.835%) [49.3mb]
microVU1: Prog Range Wrap [3ff0] [-1]
microVU1: Cached Prog = [30664] [PC=39c8] [List=8712] (Cache=80.837%) [49.3mb]
```

mVU:
![gsdx_20180415134824](https://user-images.githubusercontent.com/16314399/38781668-c37fba9e-40b6-11e8-8fb6-2c2eb098a710.png)

SuperVU:
![gsdx_20180415141043](https://user-images.githubusercontent.com/16314399/38781675-d8a737e4-40b6-11e8-8524-aa200ecba885.png)

**Comment by MrCK1** [comment]
Here's a blockdump as well: https://drive.google.com/open?id=1_HhO17wOaMqgTrMX9vQO3rlnhgkQwERW

**Comment by refractionpcsx2** [comment]
Hmm, not sure what's going on.  One disturbing thing is it actually has a branch to an area of the VU memory with nothing in it right at the end (which I suspect is what it's hitting)

[39d8] (000002ff) NOP
[39d8] (420f00c2) BAL vi15 [3ff0] 

If this is the case, I suspect some of the calculations are still wrong.

**Comment by kozarovv** [comment]
Seems like avoiding recompilation of 0x3ff0 - 0x3fff helps here a lot. Not fix root cause, but make game playable on mVU. Maybe is worth to create hack in game fixes? 

Reusing this: 
https://github.com/PCSX2/pcsx2/pull/2945/commits/167e91606cb5e2df9700b90bc83ec6e1a129da23
with those values, instead of ones required for CTR game: 
`if (isVU1 && (memcmp_mmx((u8 *)(it[0]->data), (u8 *)(mVU.regs().Micro), 0x3ff0) == 0)) {`
Seems to make game playable without any noticeable issues on microVU.

**Comment by prafullpcsx2** [comment]
Yes we are aware of this. I was using this approach to fix ihra. Ck was using different method. One of these will be added soon. Btw can you find a similar fix for shell shock nam 67? I have been trying without any success.

**Comment by kozarovv** [comment]
Looking at old Shell shock nam 67 issue here on github, i don't think is even possible.

**Comment by kozarovv** [comment]
This is quite interesting bug. Game take advantage of VU memory wrapping. All that work fine in pcsx2, excluding one BAL that always hit 0x3ff0. This is no issue per se, but due to way pcsx2 work this caused ton of unnecessary caching, and killed performance. 

Patch replace dmove a1, zero with li a1, 2. This simple trick make branch directly to 0x0 instead of 0x3ff0. To be honest it was new for me that VU support something like memory wrapping at all. Since all addresses created by that runtime always have -0x10 offset, adding 2 (2x8) make them hit precisely what should be hit. This don't have any bad side effect.

Patch is posted here: https://forums.pcsx2.net/Thread-Fixing-unplayable-games?pid=609852#pid609852

**Comment by refractionpcsx2** [comment]
Fixed in master since the #3865 merge
